### ISSUE TITLE
False positive in nullity with generic containers (`NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE`)

### ISSUE THREAD

**Issue by Proximyst** [spotbugs/spotbugs#2179]
Hi! I've found an interesting bug with false positives on generic containers, specifically futures in my case.

Minimal reproduction example:

```java
import java.util.concurrent.CompletableFuture;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class Example {

  public CompletableFuture<@Nullable String> getFutureWithNullableValue() {
    return CompletableFuture.completedFuture(null);
  }

  public @NotNull CompletableFuture<@Nullable String> getNotNullFutureWithNullableValue() {
    return CompletableFuture.completedFuture(null);
  }

  public void triggerIssue() {
    // This is not fine? It's seemingly inferring the @Nullable to affect the return value of the
    // actual method.
    this.getFutureWithNullableValue().thenAccept(s -> System.out.println("s = " + s));

    // This is luckily OK, as the future is marked not-null.
    this.getNotNullFutureWithNullableValue().thenAccept(s -> System.out.println("s = " + s));
  }
}
```

```
[INFO] --- spotbugs-maven-plugin:4.7.2.0:check (default) @ spotify-spotbugs-nullable-future-false-positive-example ---
[INFO] BugInstance size is 1
[INFO] Error size is 0
[INFO] Total bugs: 1
[ERROR] Medium: Possible null pointer dereference in com.spotify.spotifyspotbugsnullablefuturefalsepositiveexample.Example.triggerIssue() due to return value of called method [com.spotify.spotifyspotbugsnullablefuturefalsepositiveexample.Example, com.spotify.spotifyspotbugsnullablefuturefalsepositiveexample.Example] Dereferenced at Example.java:[line 20]Known null at Example.java:[line 20] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
```
(See full example project here: <https://github.com/Proximyst/spotify-spotbugs-nullable-future-false-positive-example>)

I've also reproduced it as part of the spotbugs tests, as requested by the contribution note. See: <https://github.com/Proximyst/spotbugs-fork/compare/master...Proximyst:spotbugs-fork:nullness-on-generic-container-issue>

These examples use JetBrains Annotations and CheckerFramework Annotations. I could not reproduce it with findbugs' own annotations, because they don't support `TYPE_USE`.

#2171 might also be related to this issue.

---

Legal note, because this was done at work: all submitted code is licensed under the applicable licence(s) for the project(s), but is still legally owned by Spotify AB.

**Comment by gtoison** [comment]
This should be fixed by #2502
